rollup-node.md

```diff
@@ -0,0 +1,285 @@
+[L1 attributes transaction]. The deposits (which must be similarly RLP-encoded)
+are simply copied byte-for-byte — it is the role of the [execution engine] to
+reject invalidly-formatted transactions.
+
+> **TODO** must offer some precisions on the format of deposits: sender,
+> receivers both in-tx-as-encoded, and on-L2-tx. What about the fees?
+
+The Optimistic Ethereum specific *[L1 attributes transaction]* has the following
+[EIP-2718]-compatible format: `0x7E || [block_number, timestamp, basefee]`
+where:
+
+- `0x7E` is the transaction type identifier. It is selected because transaction
+  type identifiers are currently allowed to go up to `0x7F`. Picking a high
+  identifier minimizes the risk that the identifier will be used by Ethereum in
+  the future. We don't pick `0x7F` itself in case it becomes used for a
+  variable-length encoding scheme.
+- `block_number` is the L1 block number as a 64-bit integer (4 bytes)
+- `timestamp` is the L1 block timestamp as a 64-bit integer (4 bytes)
+- `basefee` is the L1 block basefee as a 64-bit integer (4 bytes)
+
+When included in the `transactions` array, this transaction should be
+RLP-encoded in the same way as other transactions.
+
+Here is an example valid `PayloadAttributesOPV1` object, which contains an L1
+attributes transaction as well as a single deposit:
+
+```js
+{
+  timestamp: "0x61a6336f",
+  random: "0xde5dff2b0982ecbbd38081eb8f4aed0525140dc1c1d56f995b4fa801a3f2649e",
+  suggestedFeeRecipient: "0xAb5801a7D398351b8bE11C439e05C5B3259aeC9B",
+  transactions: [
+    "TODO specify L1 attribute transaction",
+    "0x02f87101058459682f0085199c82cc0082520894ab5801a7d398351b8be11c439e05c5b3259aec9b8609184e72a00080c080a0a6d217a91ea344fc09f740f104f764d71bb1ca9a8e159117d2d27091ea5fce91a04cf5add5f5b7d791a2c4663ab488cb581df800fe0910aa755099ba466b49fd69"
+  ]
+}
+```
+
+### Building the L2 block with the execution engine
+[calling-exec-engine]: #building-the-L2-block-with-the-execution-engine
+
+The Optimistic Ethereum [execution engine] is [specified in its own
+document][execution-engine-spec].
+
+This section defines how the rollup driver must interact with the execution
+engine's in order to convert [payload attributes] into L2 blocks.
+
+> **TODO** This section probably includes too much redundant details that will
+> need to be removed once the execution engine spec is up.
+
+Optimistic Ethereum's execution engine API is built upon [Ethereum's Engine API
+specification][eth-engine-api], with a couple of modifications. That
+specification builds upon [Ethereum's JSON-RPC API specification][JSON-RPC-API],
+which itself builds upon the [JSON-RPC specification][JSON-RPC].
+
+[eth-engine-api]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md
+[JSON-RPC]: https://www.jsonrpc.org/specification
+
+In particular, the [Ethereum's Engine API specification][eth-engine-api]
+specifies a [JSON-RPC] endpoint with a number of JSON-RPC (REST) routes, which
```

Comment:
```suggestion
specifies a [JSON-RPC] endpoint with a number of JSON-RPC routes, which
```
JSON-RPC is not REST. We kept the engine API design simple enough that it could be, but it's still JSON-RPC.

Reply:
Big philosophical debate, goal was just to give a quick idea, but not a hill that I'm willing to die on, will remove.